src/System.IO.Pipelines/tests/PipeReaderWriterFacts.cs

```diff
@@ -149,15 +150,62 @@ public void Dispose()
         }
 
         [Fact]
-        public async Task CompleteReaderThrowsIfReadInProgress()
+        public async Task CompleteReaderAfterFlushWithoutAdvancingDoesNotThrow()
+        {
+            await _pipe.Writer.FlushAsync();
+            ReadResult result = await _pipe.Reader.ReadAsync();
+            ReadOnlySequence<byte> buffer = result.Buffer;
+
+            _pipe.Reader.Complete();
+        }
+
+        [Fact]
+        public async Task ResetAfterCompleteReaderAndWriterWithoutAdvancingClearsEverything()
+        {
+            _pipe.Writer.WriteEmpty(4094);
+            _pipe.Writer.WriteEmpty(4094);
+            await _pipe.Writer.FlushAsync();
+            ReadResult result = await _pipe.Reader.ReadAsync();
+            ReadOnlySequence<byte> buffer = result.Buffer;
+
+            SequenceMarshal.TryGetReadOnlySequenceSegment(
+                buffer,
+                out ReadOnlySequenceSegment<byte> start,
+                out int startIndex,
+                out ReadOnlySequenceSegment<byte> end,
+                out int endIndex);
+
+            var startSegment = (BufferSegment)start;
+            var endSegment = (BufferSegment)end;
+            Assert.NotNull(startSegment.OwnedMemory);
```

Comment:
`Assert.NotEqual(default(Memory<byte>), startSegment.AvailableMemory)`

Reply:
Meh, I don't trust it 😄

Reply:
It gets reset on the next line that the OM does.